src/Microsoft.Health.Dicom.Functions/DurableFunctionsCleanup.cs

```diff
@@ -0,0 +1,86 @@
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.DurableTask;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Health.Dicom.Functions.Indexing.Configuration;
+
+namespace Microsoft.Health.Dicom.Functions
+{
+    /// <summary>
+    /// Represents the Azure Functions that perform the cleanup of completed Durable Functions used for re-indexing.
+    /// </summary>
+    public class DurableFunctionsCleanup
+    {
+        private readonly ReindexConfiguration _clientConfig;
+        private readonly ReindexErrorConfiguration _cleanupConfig;
+
+        // TODO: Leverages services for fulfilling the various activies
+        public DurableFunctionsCleanup(IOptions<IndexingConfiguration> configOptions, IOptions<ReindexErrorConfiguration> cleanupOptions)
+        {
+            EnsureArg.IsNotNull(configOptions, nameof(configOptions));
+            _clientConfig = configOptions.Value.Add;
+
+            EnsureArg.IsNotNull(cleanupOptions, nameof(cleanupOptions));
+            _cleanupConfig = cleanupOptions.Value;
+        }
+
+        // Currently an HTTP trigger, TODO: change to the appropriate trigger.
+        [FunctionName("DurableFunctionsCleanup")]
+        public async Task<String> Run(
+            [HttpTrigger(AuthorizationLevel.Function, "get", "post", Route = null)] HttpRequest req,
+            [DurableClient] IDurableOrchestrationClient client,
+            ILogger log,
+            CancellationToken hostCancellationToken)
+        {
+            EnsureArg.IsNotNull(req, nameof(req));
+            using var cancellationTokenSource = CancellationTokenSource.CreateLinkedTokenSource(hostCancellationToken, req.HttpContext.RequestAborted);
+
+            log.LogInformation("Durable Functions cleanup started.");
+
+            EnsureArg.IsNotNull(client, nameof(client));
+
+
+            EnsureArg.IsNotNull(_clientConfig, nameof(_clientConfig));
+            EnsureArg.IsNotNull(_cleanupConfig, nameof(_cleanupConfig));
+
+            // Specify conditions for Function Queries.
+            OrchestrationStatusQueryCondition condition = new OrchestrationStatusQueryCondition
+            {
+                RuntimeStatus = _cleanupConfig.RuntimeStatus,
+                CreatedTimeFrom = DateTime.MinValue,
+                CreatedTimeTo = DateTime.UtcNow.AddDays(-_cleanupConfig.DaysCompleted)
+            };
+
+            OrchestrationStatusQueryResult listOfOrchestrators =
+                await client.ListInstancesAsync(new OrchestrationStatusQueryCondition(), cancellationTokenSource.Token);
```

Comment:
You'll want to pass the `OrchestrationStatusQueryCondition` you just created

Reply:
yup! this was for testing purposes